File: view/src/main/java/com/seniorlibs/view/widget/MyViewGroup.java
package com.seniorlibs.view.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.util.Log;
import android.view.ViewGroup;

/**
 * 如果有背景色，走onDraw()方法；如果没有背景色，不走onDraw()方法
 */
public class MyViewGroup extends ViewGroup {
    private static final String TAG = "MyViewGroup";
    public MyViewGroup(Context context) {
        super(context);
    }
    public MyViewGroup(Context context, AttributeSet attrs) {
        super(context, attrs);
    }
    @Override
    protected void onLayout(boolean changed, int l, int t, int r, int b) {
    }
    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        Log.d(TAG, "onDraw:执行了");
    }
    @Override
    protected void dispatchDraw(Canvas canvas) {
        super.dispatchDraw(canvas);
        Log.d(TAG, "dispatchDraw:执行了");
    }
}

File: event/src/main/java/com/seniorlibs/event/dispatch/MyViewGroup.kt
package com.seniorlibs.event.dispatch

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout
import com.seniorlibs.event.utils.EventUtils
import com.seniorlibs.baselib.utils.LogUtils

class MyViewGroup(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
    private val TAG = "DispatchActivityTag"

    override fun dispatchTouchEvent(event: MotionEvent): Boolean {
        LogUtils.i(TAG,"ViewGroup ：dispatchTouchEvent --> ${EventUtils.getAction(event.action)}")
        return super.dispatchTouchEvent(event)
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        LogUtils.i(TAG, "ViewGroup ：onTouchEvent --> ${EventUtils.getAction(event.action)}")
        return super.onTouchEvent(event)
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        LogUtils.e(TAG,"ViewGroup ：onInterceptTouchEvent --> ${EventUtils.getAction(event.action)}")
        return super.onInterceptTouchEvent(event)
    }
}